testsuite/tests/langkit_support/diagnostics_output/main.adb

```diff
@@ -0,0 +1,70 @@
+with Ada.Exceptions; use Ada.Exceptions;
+with Ada.Strings.Wide_Wide_Fixed;
+with Ada.Text_IO;    use Ada.Text_IO;
+with Ada.Wide_Wide_Text_IO;
+
+with Langkit_Support.Diagnostics.Output;
+use Langkit_Support.Diagnostics;
+
+with Langkit_Support.Slocs;   use Langkit_Support.Slocs;
+with Langkit_Support.Symbols; use Langkit_Support.Symbols;
+with Langkit_Support.Text;    use Langkit_Support.Text;
+with Langkit_Support.Token_Data_Handlers;
+use Langkit_Support.Token_Data_Handlers;
+
+with Ada.Containers.Vectors;
+
+procedure Main is
+   package WWTIO renames Ada.Wide_Wide_Text_IO;
+
+   package Text_Vectors is new Ada.Containers.Vectors
+     (Positive, Unbounded_Text_Type);
+
+   type Simple_Buffer is new Text_Buffer_Ifc with record
+      Lines : Text_Vectors.Vector;
+   end record;
+
+   overriding function Get_Line
+     (Self : Simple_Buffer; Line_Number : Positive) return Text_Type
+   is
+   begin
+      return To_Text (Self.Lines.Element (Line_Number));
+   end;
+
+   function Create_From_File (File_Path : String) return Simple_Buffer is
+      FT : WWTIO.File_Type;
+      Ret : Simple_Buffer;
+   begin
+      WWTIO.Open (FT, WWTIO.In_File, File_Path);
+
+      loop
+         declare
+            Line : Text_Type := WWTIO.Get_Line (FT);
+         begin
+            Ret.Lines.Append (To_Unbounded_Text (Line));
+         end;
+      end loop;
+
+   exception
+      when WWTIO.End_Error =>
+         WWTIO.Close (FT);
+         return Ret;
+   end Create_From_File;
+   
+   B : Simple_Buffer := Create_From_File ("main.adb");
+
+begin
+   Put_Line ("Regular diagnostic case");
+   Output.Print_Diagnostic
+     (Diagnostic'
+       (Sloc_Range => Source_Location_Range'(31, 31, 14, 21),
```

Comment:
Please align arrows.

Btw, if you find it more convienient, you can use the `Value` function from `Langkit_Support.Slocs`: `Value (String'("31:14-31:21"))`